Periodic_2_triangulation_2/include/CGAL/draw_periodic_2_triangulation_2.h

```diff
@@ -23,255 +26,196 @@
-         it != p2t2.periodic_segments_end(it_type); it++)
-    {
-      compute_edge(it);
-    }
+template <typename BufferType = float, class P2T2>
+void add_in_graphic_buffer(const P2T2 &p2t2, CGAL::Graphic_buffer<BufferType> &graphic_buffer) {
 
-    for (Periodic_triangle_iterator it =
-             p2t2.periodic_triangles_begin(it_type);
-         it != p2t2.periodic_triangles_end(it_type); it++)
-    {
-      compute_face(it);
-    }
+  Drawing_functor<P2T2,typename P2T2::Periodic_triangle_iterator,
+                  typename P2T2::Periodic_triangle_iterator,
+                  typename P2T2::Periodic_triangle_iterator> drawing_functor;
 
-    if(m_domain){
-      // Compute the (9-sheet covering space) domain of the periodic triangulation
-      compute_domain();
-    }
-  }
+  drawing_functor.colored_face = [](const P2T2&,
+                      typename P2T2::Periodic_triangle_iterator) -> bool
+  { return true; };
 
-  virtual void keyPressEvent(QKeyEvent *e)
-  {
-    // Test key pressed:
-    //    const ::Qt::KeyboardModifiers modifiers = e->modifiers();
-    //    if ((e->key()==Qt::Key_PageUp) && (modifiers==Qt::NoButton)) { ... }
-
-    // Call: * compute_elements() if the model changed, followed by
-    //       * redraw() if some viewing parameters changed that implies some
-    //                  modifications of the buffers
-    //                  (eg. type of normal, color/mono)
-    //       * update() just to update the drawing
-
-    // Call the base method to process others/classicals key
-    const ::Qt::KeyboardModifiers modifiers = e->modifiers();
-    if (e->text()=="1")
-    {
-      m_display_type = Display_type::UNIQUE;
-      displayMessage(QString("Display type= UNIQUE"));
-      compute_elements();
-      redraw();
-    } else if (e->text()=="2")
-    {
-      m_display_type = Display_type::UNIQUE_COVER_DOMAIN;
-      displayMessage(QString("Display type= UNIQUE_COVER_DOMAIN"));
-      compute_elements();
-      redraw();
-    } else if (e->text()=="3")
-    {
-      m_display_type = Display_type::STORED;
-      displayMessage(QString("Display type= STORED"));
-      compute_elements();
-      redraw();
-    } else if (e->text()=="4")
-    {
-      m_display_type = Display_type::STORED_COVER_DOMAIN;
-      displayMessage(QString("Display type= STORED_COVER_DOMAIN"));
-      compute_elements();
-      redraw();
-    } else if ((e->key() == ::Qt::Key_D) && (modifiers == ::Qt::NoButton))
-    {
-      m_domain=!m_domain;
-      displayMessage(QString("Draw domain=%1.").arg(m_domain?"true":"false"));
-      compute_elements();
-      redraw();
-    } else {
-      Base::keyPressEvent(e);
-    }
-  }
+  // TODO: I think we need to add std::function like this:
+  // drawing_functor.face_color =  [] (const P2T2&,
+  //                          typename P2T2::Periodic_triangle_iterator,
+  // int R, int G, int B)  -> CGAL::IO::Color
+  // {
+  //     return CGAL::IO::Color(R, G, B);
+  // };  // What do you think?
```

Comment:
What do you think? @gdamiand

Reply:
Yes, but I think we have multiple changes for coloring, e.g. [Line:71](https://github.com/CGAL/cgal/blob/7404eb629fda75cefe6786c2ca625b05d036fe7c/Periodic_2_triangulation_2/include/CGAL/draw_periodic_2_triangulation_2.h#L71), [Line:85](https://github.com/CGAL/cgal/blob/7404eb629fda75cefe6786c2ca625b05d036fe7c/Periodic_2_triangulation_2/include/CGAL/draw_periodic_2_triangulation_2.h#L85), [Line(s)111:114](https://github.com/CGAL/cgal/blob/7404eb629fda75cefe6786c2ca625b05d036fe7c/Periodic_2_triangulation_2/include/CGAL/draw_periodic_2_triangulation_2.h#L111). right?

Reply:
Only the first color is a face color (line 71). The second color line 85 is for segments (you can keep the constant) and the third color is also for segments, and here again you can keep a constant (like in the current code).

Reply:
I see, Thank you!